design-patterns/src/main/java/proxy/SecureProxy.java

```diff
@@ -0,0 +1,34 @@
+package proxy;
+
+import java.nio.file.AccessDeniedException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class SecureProxy extends Proxy{
+    List<String> allowedUsers = new ArrayList<>();
+    public SecureProxy(){
```

Comment:
please, don't put any logic inside the constructor, it is a bad idea in general

in this case you can instantiate an allowance list directly